atoms/userProfile.ts

```diff
@@ -0,0 +1,9 @@
+import {atom} from "recoil";
+import {FirebaseUserProfile} from "../utils/types/firebase";
+
+// userProfileState is an Atom, a global state
+// that can be accessed by any component
+export const userProfileState = atom<FirebaseUserProfile | {}>({
+    key: "userProfileState", // unique ID of the global state atom
+    default: {}, // default value is empty
+});
```

Comment:
Would we need to fetch that in real time when needed?

Reply:
Most likely. I was thinking we'd need the current user so that we could access the entry id for the user collection since it's used everywhere for calls.

I just solved this by storing the user and profile id within the profile data itself so it's always accessible in global state.